### ISSUE TITLE
App crash - "Value does not fall within the expected range."

### ISSUE THREAD

**Issue by mjracca** [Quobject/EngineIoClientDotNet#10]
I'm having this random exception which maskes my app crash

![image](https://cloud.githubusercontent.com/assets/3976257/8383486/122dfade-1c11-11e5-8c89-536cf6729166.png)

> **Value does not fall within the expected range.**
> index and count exceed length of list

```
at Quobject.Collections.Immutable.ImmutableList1.CheckRange(Int32 idx, Int32 count) at Quobject.Collections.Immutable.ImmutableList1.RemoveRange(Int32 index, Int32 count)
at Quobject.EngineIoClientDotNet.Client.Socket.OnDrain()
at Quobject.EngineIoClientDotNet.Client.Socket.EventDrainListener.Quobject.EngineIoClientDotNet.ComponentEmitter.IListener.Call(Object[] args)
at Quobject.EngineIoClientDotNet.ComponentEmitter.Emitter.Emit(String eventString, Object[] args)
at Quobject.EngineIoClientDotNet.Client.Transports.WebSocket.b__0()
at Quobject.EngineIoClientDotNet.Thread.EasyTimer.<>c__DisplayClass1.b__0()
at System.Runtime.CompilerServices.TaskAwaiter.<>c__DisplayClass0.b__1()
at System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()
at System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation.<>c__DisplayClass0.<_cctor>b__1(Object state)
at System.Threading.WinRTSynchronizationContext.Invoker.InvokeCore()
```

**Comment by mattqs** [comment]
What version of EnginIoClientDotNet are you using? 

There was a bug fix in 0.9.14 https://github.com/Quobject/SocketIoClientDotNet/issues/18 that should prevent this exception.

**Comment by megamingus** [comment]
We are currently using version 0.9.17

**Comment by mattqs** [comment]
Fixed in 0.9.18.
